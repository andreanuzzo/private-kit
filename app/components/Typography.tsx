import React from 'react';
import { Text, TextStyle } from 'react-native';

import { useLanguageDirection } from '../locales/languages';

import { Typography as TypographyStyles } from '../styles';

type TypographicUse =
  | 'headline1'
  | 'headline2'
  | 'headline3'
  | 'headline4'
  | 'headline5'
  | 'headline6'
  | 'headline7'
  | 'body1'
  | 'body2'
  | 'body3'
  | 'body4';

interface TypographyProps {
  use?: TypographicUse;
  style?: TextStyle;
  testID?: string;
  children: JSX.Element | string;
  onPress?: () => void;
}

export const Typography = ({
  use = 'body1',
  style,
  testID,
  children,
  onPress,
}: TypographyProps): JSX.Element => {
  const writingDirection = useLanguageDirection();

  const useToStyle = () => {
    switch (use) {
      case 'headline1': {
        return TypographyStyles.header1;
      }
      case 'headline2': {
        return TypographyStyles.header2;
      }
      case 'headline3': {
        return TypographyStyles.header3;
      }
      case 'headline4': {
        return TypographyStyles.header4;
      }
      case 'headline5': {
        return TypographyStyles.header5;
      }
      case 'headline6': {
        return TypographyStyles.header6;
      }
      case 'headline7': {
        return TypographyStyles.header7;
      }
      case 'body1': {
        return TypographyStyles.mainContent;
      }
      case 'body2': {
        return TypographyStyles.secondaryContent;
      }
      case 'body3': {
        return TypographyStyles.tertiaryContent;
      }
      case 'body4': {
        return TypographyStyles.quaternaryContent;
      }
    }
  };

  const textStyle = useToStyle();

  return (
    <Text
      onPress={onPress}
      style={[textStyle, { writingDirection }, style]}
      testID={testID}>
      {children}
    </Text>
  );
};
