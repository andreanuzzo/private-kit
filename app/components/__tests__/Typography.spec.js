import { render } from '@testing-library/react-native';
import React from 'react';

import { Typography } from '../Typography';

it('headline1 is large and bold', () => {
  const { asJSON } = render(<Typography use='headline1'>headline1</Typography>);

  expect(asJSON()).toMatchSnapshot();
});

it('body1 is regular', () => {
  const { asJSON } = render(<Typography use='body1'>body1</Typography>);

  expect(asJSON()).toMatchSnapshot();
});
