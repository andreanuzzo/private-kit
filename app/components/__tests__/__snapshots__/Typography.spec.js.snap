// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body1 is regular 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#2e2e2e",
          "fontFamily": "IBMPlexSans",
          "fontSize": 17,
          "lineHeight": 24,
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    body1
  </Text>
</View>
`;

exports[`headline1 is large and bold 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1f2c9b",
          "fontFamily": "IBMPlexSans-Bold",
          "fontSize": 52,
          "fontWeight": "500",
          "lineHeight": 50,
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    headline1
  </Text>
</View>
`;
